synapse/storage/data_stores/main/events.py

```diff
@@ -908,6 +910,8 @@ def _update_metadata_tables_txn(
                 # Remove from relations table.
                 self._handle_redaction(txn, event.redacts)
 
+                self._delete_event_expiry_txn(txn, event.redacts)
```

Comment:
I'd hesitate to do this. it's harmless if the expiry runs after the event is redacted, and there are problems with this area of code.